### ISSUE TITLE
find_local_maxima() gives different results in python2 vs python3

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#151]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/151) by Dan Becker (Bitbucket: [danbek](https://bitbucket.org/danbek), GitHub: [danbek](https://github.com/danbek)).**

----------------------------------------

Using the attached npz file, this code gives different results in python2 vs python3.

```
#!python
import numpy as np
import mass
ph = np.load('py3bug.npz')['ph']
mass.algorithms.find_local_maxima(ph, 50)
```
Haven't had a chance to investigate this yet, but I'm happy to do so next week if no one gets to it first.

**Comment by joefowler** [comment]
**Original comment by Dan Becker (Bitbucket: [danbek](https://bitbucket.org/danbek), GitHub: [danbek](https://github.com/danbek)).**

----------------------------------------

The source of the problem is this line in `find_local_maximum()`:
```
tx = np.fft.rfftfreq(n, (lo - hi) / n)
```
If the pulse heights are integers, and your `gaussian_fwhm` value is an integer (50 in this case), then `lo`, `hi`, and `n` will all be integers. And then in python 2 the result of `(lo - hi)/n` will also be an integer, but in python3 it will be a float.

I suspect no one has run into this problem because they don't usually pass in pulse heights as integers (I'm using `p_pulse_height` to do a "pre-calibration" to identify the region over which to calculate average pulses).

I think an easy fix is to force lo and hi to always be floats, but the reason that I discovered this at all is that my "pre-calibration" was failing under python3 for some devices, and I first want to understand why this happens with the python3 behavior (which strikes me as more correct).

**Comment by joefowler** [comment]
**Original comment by Dan Becker (Bitbucket: [danbek](https://bitbucket.org/danbek), GitHub: [danbek](https://github.com/danbek)).**

----------------------------------------

OK, I now understand what was going on with my use of this, and I'm quite certain that the solution is to ensure that the quantity `(lo - hi)/n` is always a floating point number. I'll submit a pull request to fix this soon.

**Comment by joefowler** [comment]
**Original comment by Dan Becker (Bitbucket: [danbek](https://bitbucket.org/danbek), GitHub: [danbek](https://github.com/danbek)).**

----------------------------------------

Make find_local_maxima() work with integer pulse heights in python3

Fixes #151

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Removing milestone: v0.7 (automated comment)
